aleph-client/src/contract/convertible_value.rs

```diff
@@ -104,3 +104,73 @@ where
         bail!("Expected {:?} to be an Ok(_) or Err(_) tuple.", value);
     }
 }
+
+impl TryFrom<ConvertibleValue> for String {
+    type Error = anyhow::Error;
+
+    fn try_from(value: ConvertibleValue) -> std::result::Result<String, Self::Error> {
+        if let Value::Seq(seq) = value.0 {
+            let mut bytes: Vec<u8> = Vec::with_capacity(seq.len());
+            for el in seq.elems() {
+                if let Value::UInt(byte) = *el {
+                    if byte > u8::MAX as u128 {
+                        bail!("Expected number <= u8::MAX but instead got: {:?}", byte)
+                    }
+                    bytes.push(byte as u8);
+                } else {
+                    bail!("Failed parsing `ConvertibleValue` to `String`. Expected `Value::UInt` but instead got: {:?}", el);
+                }
+            }
+            String::from_utf8(bytes).context("Failed parsing bytes to UTF-8 String.")
+        } else {
+            bail!("Failed parsing `ConvertibleValue` to `String`. Expected `Seq(Value::UInt)` but instead got: {:?}", value);
+        }
+    }
+}
+
+auto trait NotEq {}
+// We're basically telling the compiler that there is no instance of NotEq for `(X,X)` tuple.
+// Or put differently - that you can't implement `NotEq` for `(X,X)`.
+impl<X> !NotEq for (X, X) {}
+
+impl<T> TryFrom<ConvertibleValue> for Option<T>
+where
+    T: TryFrom<ConvertibleValue, Error = anyhow::Error> + Debug,
+    // We will derive this impl only when `T != ConvertibleValue`.
+    // Otherwise we will get a conflict with generic impl in the rust `core` crate.
+    (ConvertibleValue, T): NotEq,
+{
+    type Error = anyhow::Error;
+
+    fn try_from(value: ConvertibleValue) -> std::result::Result<Option<T>, Self::Error> {
+        if let Value::Tuple(tuple) = &value.0 {
```

Comment:
to reduce nesting (here and in other similar places
```rust
let tuple = match value.0 {
 Value::Tuple(tuple) => tuple,
 _ => bail!(...),
};
...
```

Reply:
I think I tried that, but it makes it so you have to duplicate the `bail!` in the case where we now do `_ => ()`. I decided this was better than having the `bail!` twice.

Reply:
I went with the suggestion though. It's true that it adds a `bail!` call but it's a different one -- i.e. different error message and I agree with @kostekIV that returning early and decreasing nesting improves readability.